joint_trajectory_controller/test/test_trajectory_controller.cpp

```diff
@@ -581,54 +656,31 @@ TEST_F(TestTrajectoryController, test_jumbled_joint_order) {
     trajectory_publisher_->publish(traj_msg);
   }
 
-  // update for 0.5 seconds
-  const auto start_time = rclcpp::Clock().now();
-  const rclcpp::Duration wait = rclcpp::Duration::from_seconds(0.5);
-  const auto end_time = start_time + wait;
-  while (rclcpp::Clock().now() < end_time) {
-    test_robot_->read();
-    traj_controller_->update();
-    test_robot_->write();
-  }
+  traj_controller_->wait_for_trajectory(executor);
+  // update for 0.25 seconds
+  updateController(rclcpp::Duration::from_seconds(0.25));
 
   EXPECT_NEAR(1.0, test_robot_->pos1, COMMON_THRESHOLD);
   EXPECT_NEAR(2.0, test_robot_->pos2, COMMON_THRESHOLD);
   EXPECT_NEAR(3.0, test_robot_->pos3, COMMON_THRESHOLD);
-
-  executor.cancel();
 }
 
+/**
+ * @brief test_partial_joint_list Test sending trajectories with a subset of the controlled joints
+ */
 TEST_F(TestTrajectoryController, test_partial_joint_list) {
```

Comment:
Could you please add a pretty much identical test where we check the behaviour of the controller to a partial list with the feature turned off?